lib/AST/Decl.cpp

```diff
@@ -3485,7 +3485,7 @@ SourceRange TypeAliasDecl::getSourceRange() const {
     return { TypeAliasLoc, TrailingWhereClauseSourceRange.End };
   if (UnderlyingTy.hasLocation())
     return { TypeAliasLoc, UnderlyingTy.getSourceRange().End };
-  return { TypeAliasLoc, getNameLoc() };
+  return {TypeAliasLoc, EndLoc};
```

Comment:
This implementation of `getSourceRange()` implies that `TypeAliasDecl::EndLoc` is not truly the end of the source range.  We should either make it so `EndLoc` is always the end of the source range, or we should rename it to something that indicates more clearly what it is.

Reply:
Yes, I agree. I'll change it and run the tests.

Reply:
OK, what I called "EndLoc" for the TypeAliasDecl is really the end of the type when parseType decides NOT to return a parsed type. I could call it "EndOfUnparsableType"

Reply:
I think I'll called it `TypeEndLoc`.